### PR TITLE
Release 2.0.0

### PR DESCRIPTION
**Pull request by pv711**

### CODE REVIEW

File: engine/src/main/battlecode/common/GameConstants.java [hunk 1]
```diff
@@ -99,7 +102,7 @@ public class GameConstants {
     public static final int DISTANCE_SQUARED_FROM_HEADQUARTER = 9;
 
     /** The discount factor on the amount of damage a carrier can do based on their capacity */
-    public static final float CARRIER_DAMAGE_FACTOR = 0.2f;
+    public static final float CARRIER_DAMAGE_FACTOR = 1.25f;
```
Comment (acrantel) [inline engine/src/main/battlecode/common/GameConstants.java:105]: idk how engine uses this but assumign this is intended?

File: engine/src/main/battlecode/common/RobotType.java [hunk 2]
```diff
@@ -55,15 +55,15 @@ public enum RobotType {
...
-    BOOSTER         (  0,   0, 150, 140,  25, 40,   0, -1, 20, 10000),
+    BOOSTER         (  0,   0, 150, 140,  25, 400,   0, -1, 20, 10000),
     //               BCA  BCM  BCE   AC   MC  HP  DMG  AR  VR     BL
     /**
      * Amplifiers are the key to communication for your army. They enable 
      * robots within a range to write messages to their faction’s shared array.
      *
      * @battlecode.doc.robottype
      */
-    AMPLIFIER       ( 40,  40,   0,   -1,  15, 40,   0, -1, 34,  7500)
```
Comment (acrantel) [inline engine/src/main/battlecode/common/RobotType.java:66]: looks like amplifier health used to be 40, not 400 (we should change this in the specs changelog)

File: engine/src/main/battlecode/common/GameConstants.java (hunk 1 above)
Comment (pv711) [reply engine/src/main/battlecode/common/GameConstants.java:105]: we multiplied everything by 10 so really this is a reduction from 2 -> 1.25

File: engine/src/main/battlecode/common/RobotType.java (hunk 2 above)
Comment (pv711) [reply engine/src/main/battlecode/common/RobotType.java:66]: yeah again multiply by 10 is null op but feel free to change if you think it makes more sense

File: engine/src/main/battlecode/world/InternalRobot.java [hunk 3]
```diff
@@ -392,14 +392,21 @@ private int getDamage() {
...
+     * Attacks another location (launcher).
      * 
      * @param loc the location of the bot
      */
     public void attack(MapLocation loc) {
         InternalRobot bot = this.gameWorld.getRobot(loc);
-        if (!(bot == null)) {
+        if (bot == null || bot.getTeam() == this.getTeam() || bot.getType() == RobotType.HEADQUARTERS) {
+            // If robot is null, of your team, or a hq do no damage, otherwise do damage
+            this.getGameWorld().getMatchMaker().addAction(getID(), Action.LAUNCH_ATTACK, -locationToInt(loc) - 1);
```
Comment (acrantel) [inline engine/src/main/battlecode/world/InternalRobot.java:408]: i'm not sure what this function does ;; but it looks like we're launching attacks on headquarters or bots of our own team?

**Pull request merged**